📖 .github: improve kubernetes bump issue template (#9209)
kubernetes-sigs/cluster-api: .github/ISSUE_TEMPLATE/kubernetes_bump.md

    * `test/*`: search for occurrences of the previous Kubernetes version
    * `Tiltfile`
  * Ensure the latest available kind version is used (including the latest images for this kind release)

>     * Add new images in the [kind mapper.go](https://github.com/kubernetes-sigs/cluster-api/blob/48ae58e51f9723ab7b9635d0e05ee54c4843707a/test/infrastructure/kind/mapper.go#L79)

Reviewer: can we also clarify please where to get kind image sha from?

    * Set new default image for the [test framework](https://github.com/kubernetes-sigs/cluster-api/blob/48ae58e51f9723ab7b9635d0e05ee54c4843707a/test/framework/bootstrap/kind_provider.go#L40)
  * Verify the quickstart manually
  * Bump `InitWithKubernetesVersion` and `WorkloadKubernetesVersion` in `clusterctl_upgrade_test.go`

>     * Note: Do not bump for upgrades which are using older clusterctl versions.

Reviewer: ```suggestion
    * Note: Do not bump for upgrades using older clusterctl versions.
```

    * `test/*`: search for occurrences of the previous Kubernetes version
    * `Tiltfile`
  * Ensure the latest available kind version is used (including the latest images for this kind release)
    * Add new images in the [kind mapper.go](https://github.com/kubernetes-sigs/cluster-api/blob/48ae58e51f9723ab7b9635d0e05ee54c4843707a/test/infrastructure/kind/mapper.go#L79).
      * See the [kind releases page](https://github.com/kubernetes-sigs/kind/releases) for the list of released images.
    * Set new default image for the [test framework](https://github.com/kubernetes-sigs/cluster-api/blob/48ae58e51f9723ab7b9635d0e05ee54c4843707a/test/framework/bootstrap/kind_provider.go#L40)
  * Verify the quickstart manually
  * Bump `InitWithKubernetesVersion` and `WorkloadKubernetesVersion` in `clusterctl_upgrade_test.go`

>     * Note: Do not bump for upgrades using older clusterctl versions.

Reviewer 2: ```suggestion
    * Note: Only bump for Cluster API versions that will support the new Kubernetes release.
```
I think?

  * Prior art: #9160
* [ ] Ensure the jobs are adjusted to provide test coverage according to our [support policy](https://cluster-api.sigs.k8s.io/reference/versions.html#supported-kubernetes-versions):
  * For the main branch:
    * periodics:

>       * Bump `KUBERNETES_VERSION_MANAGEMENT` of the `e2e-mink8s` job to the new minimum supported management cluster version.

Reviewer 2: I think this is also for the presubmit

Author: Great point. I only did it for the periodics 😭

      * Drop the oldest upgrade job as the oldest Kubernetes minor version is now out of support.
      * Add new upgrade job which upgrades from the previous to the new Kubernetes version.
    * periodics & presubmits:
      * Bump `KUBEBUILDER_ENVTEST_KUBERNETES_VERSION` of the `test-mink8s` jobs to the new minimum supported management cluster version.

>       * Adjust the `-latest` upgrade job to upgrade from the new Kubernetes and upgrade to the next Kubernetes version.

Reviewer 2: ```suggestion
      * Adjust the `-latest` upgrade job to upgrade from the new Kubernetes to the next Kubernetes version.
```
